### DISCUSSION TITLE
[dashboard] grid granularity

### DISCUSSION THREAD

**Discussion by kamalkeshavani-aiinside** [apache/superset#18499] [Ideas]
**Is your feature request related to a problem? Please describe.**
I would like to increase the granularity of the grid on dashboard layout.

**Describe the solution you'd like**
I can see an old PR for the same, but not sure if this was lost in code refactoring. I can't see this feature in master anymore, and since these files do not exist anymore in code, I am not sure what happened to this PR.
https://github.com/apache/superset/pull/4416

**Describe alternatives you've considered**

**Additional context**
It would be great if we can get this in 1.3.

**Comment by junlincc** [comment]
Not sure about the timeline, but we should bring it back. 🙏 thanks for filing

**Comment by villebro** [comment]
I don't remember having seen this increased granularity on the v2 dashboards. Is this perhaps something that was knowingly removed in the new dashboard layout? Looking at docs here: https://gist.github.com/williaster/bad4ac9c6a71b234cf9fc8ee629844e5 , especially 
![image](https://user-images.githubusercontent.com/33317356/129138805-a537409c-a8ab-417b-8fd9-263e13456842.png)

Ping @mistercrunch @graceguo-supercat

**Comment by junlincc** [comment]
removing the priority label until we have more context/.

**Comment by graceguo-supercat** [comment]
Hi @junlincc @villebro what context do you need for dashboard grid granularity?

**Comment by villebro** [comment]
@graceguo-supercat I think it would be important to understand why the grid was expanded to 48 columns in https://github.com/apache/superset/pull/4416 but later was reduced to 12 columns in Dashboard v2 as per the screenshot above in https://gist.github.com/williaster/bad4ac9c6a71b234cf9fc8ee629844e5 . Was the more granular grid found to be unintuitive or problematic somehow?

**Comment by graceguo-supercat** [comment]
because of the very tiny grid, you have to drag the mouse or touch pad tiny bit by bit to align items in a row or column, it turn out to be very annoying :)

**Comment by kamalkeshavani-aiinside** [comment]
We already have very tiny grid for vertical size, so I thought making it available for horizontal size should be also ok.

But if we want to avoid tiny horizontal grid as default, here is my suggestion(haven't checked full code yet):
1. Currently, we store the width of every chart as number of columns in dashboard since number of columns are fixed across all dashboards.
2. We can add an extra KV for new dashboards to store column count, default as 12.
3. Absence of this key(for existing dashboard) can be assumed as column count 12 and added when existing dashboard is edited.
3. Need to create a new UI where user can select the column count(only from multiples of 12?) for any dashboard, which will be saved in the new key.
4. Change of this column count in dashboard will update the chart width(round down when exact not possible, e.x, width of 7 columns with column count 24, will become 3 when column count changed to 12 and so on)

Such implementation will not require db upgrade like in https://github.com/apache/superset/pull/4416.
I can try to contribute for python code, but will need help for UI changes.
